File: HospitalAPI/API/Services/FieldService/FieldService.cs
using API.Data;
using API.Data.Models;

namespace API.Services
{
    public class FieldService : IFieldService
    {
        private readonly HospitalDB _database;

        public FieldService(HospitalDB database)
        {
            this._database = database;
        }

        public IQueryable<Field> ListFields()
        {
            return this._database
                    .Field;
        }
    }
}

File: HospitalWebAngular/src/app/services/field/field.service.ts
import { Injectable } from '@angular/core';
import { HttpService } from '@services/http/http.service';
import { APIResponse } from '@services/http/http.types';
import { map, Observable, tap } from 'rxjs';
import { Field } from './field.model';

@Injectable({
  providedIn: 'root'
})
export class FieldService {
  private readonly _api: string;

  constructor(
    private httpService: HttpService
  ) { 
    this._api = "fields";
  }

  public list(): Observable<Field[]> {
    return this.httpService.get(this._api)
      .pipe<Field[]>(
        map((response: APIResponse) => response?.data.map((item: any) => new Field(item)))
      );
  }
}
